### ISSUE TITLE
Allow downstream projects to disable the enforcement of 64 bit precision

### ISSUE THREAD

**Issue by diegoferigo** [ami-iit/jaxsim#109]
So far, in JaxSim we always aimed to obtain results that could be comparable with those computed by third-party multibody dynamic libraries like [robotology/idyntree](https://github.com/robotology/idyntree/). For this reason, we enforce jax to run using [64 bit precision](https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#double-64bit-precision).

https://github.com/ami-iit/jaxsim/blob/4fd2032c74de495cd59e1f6999e1f7c3c9ec2ff8/src/jaxsim/__init__.py#L6-L20

This is necessary in order to compute quantities with enough accuracy that enables a comparison with small tolerances against third-party libraries (check #106 for more details).

However, especially on GPUs, using 64 bit precision may lead to excessive memory usage and/or could significantly decrease runtime performance. Therefore, we should:

- Assess if simulating with 32 bit precision maintains the same stability of using 64 bit.
- While maintaining 64 bit precision as default choice, allow downstream projects to disable the 64 bit enforcement.

**Comment by diegoferigo** [comment]
> While maintaining 64 bit precision as default choice, allow downstream projects to disable the 64 bit enforcement.

I just realized that with the current logic, 32 bit precision can be already enforced downstream by exporting `JAX_ENABLE_X64=0`. If this environment variable is set to 0, JaxSim does not configure JAX to run with 64 bit precision when it is first imported in an application.

> Assess if simulating with 32 bit precision maintains the same stability of using 64 bit.

As expected, the default pytest tolerances are too strict when comparing quantities computed with iDynTree against the equivalent computed with JaxSim in 32 bit. Our test suite fails in this case. This doesn't mean that running code with 32 bit precision is bugged, it only means that -as expected- we lose accuracy. There are many applications, especially those highly parallel running on GPUs and  TPUs, that would value more a much lower memory usage over the fidelity of the simulation (that already accounts of many approximations of real-world physics).

---

This being said, I think we can close this issue.
